### ISSUE TITLE
ARM build failing due to padding warning in System.Net.Security.Native

### ISSUE THREAD

**Issue by prajwal-aithal** [dotnet/corefx#7342]
I encounter a `-Wpadded` warning when building corefx for Linux arm32 environment (using the `arm cross` option in the build script). The build fails due to the presence of `-Werror` flag.
```
In file included from /home/prajwal/dotnet/corefx/src/Native/System.Net.Security.Native/pal_gssapi.cpp:7:
/home/prajwal/dotnet/corefx/src/Native/System.Net.Security.Native/pal_gssapi.h:42:8: error: 
      padding size of 'PAL_GssBuffer' with 4 bytes to alignment boundary [-Werror,-Wpadded]
struct PAL_GssBuffer
       ^
1 error generated.
```
This occurs due to the size of uint8_t pointer being 4 bytes on Linux arm32.

I am able to suppress this warning using the `-Wno-padded` warning in the CMakeLists file. Is this a valid fix (suppressing all future padding warnings)?

**Comment by prajwal-aithal** [comment]
cc: @myungjoo  @jasonwilliams200OK

**Comment by myungjoo** [comment]
If -Wno-padded is activated only for systems not requiring 8byte-alignment including ARM32, it should be fine.

**Comment by jasonwilliams200OK** [comment]
This seems to be duplicate of #7090, which was fixed by fa95544. Are you compiling with latest master?

**Comment by prajwal-aithal** [comment]
@jasonwilliams200OK checked with latest master. Warning is still present.

fa95544831e253d2b0a869b83dec13fc24420e0a fixed the loss of precision (due to implicit conversion) error. I guess while reporting @myungjoo had already disabled `-Werror` in his local machine.

**Comment by jasonwilliams200OK** [comment]
@prajwal-aithal, try swapping positions:

```c
struct PAL_GssBuffer
{
    uint64_t length;
    uint8_t* data;
};
```

to 

```c
struct PAL_GssBuffer
{
    uint8_t* data;
    uint64_t length;
};
```

If that does not fix the error or causes some other error, then I think options are to disable this error (`-Wno-padded`) or disable the warning inline (like ba88726):

```c
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wpadded"
// target code
#pragma clang diagnostic pop
```

**Comment by myungjoo** [comment]
@jasonwilliams200OK @prajwal-aithal This issue is not resolved by #7090 / fa95544. I've been suppressing this error manually.

**Comment by prajwal-aithal** [comment]
@jasonwilliams200OK Swapping the order of struct members does not remove this warning. I will go with the inline disabling of this warning for that piece of code. Thank you.
